lib/FrameManager.js

```diff
@@ -566,8 +566,15 @@ class Frame {
    * @return {!Promise}
    */
   waitFor(selectorOrFunctionOrTimeout, options = {}, ...args) {
-    if (helper.isString(selectorOrFunctionOrTimeout))
-      return this.waitForSelector(/** @type {string} */(selectorOrFunctionOrTimeout), options);
+    const xpathPattern = /^\//;
```

Comment:
let's check for double-slash here, otherwise it gonna be a breaking change.

Consider the following usecase: 
```js
const crazySelector = '/* comment */div';
await page.waitFor(crazySelector);
```

Reply:
would that work?

Reply:
crazy, fixed.